### DISCUSSION TITLE
Ubuntu 24.04 LTS doesnt allow Phoenix

### DISCUSSION THREAD

**Discussion by ghost** [phcode-dev/phoenix#1644] [General]
Hi there, 

I upgraded my ubuntu installation this morning to Ubuntu 24.04 LTS and phoenix stopped working. It is not currently supported by this installation script. Any workaround for the same. Thank you and it is a great addition to the web code as it allows for immediate checks.

This is what i am getting. 
```
(base) ➜  ~ wget -qO- https://updates.phcode.io/linux/installer.sh | bash
Using temporary directory /tmp/tmp.y2bvxw0tSX for processing
Fetching the latest release information from phcode-dev/phoenix-desktop...
Current GLIBC version: 2.39
Searching for a compatible binary...
Found a new best match: https://github.com/phcode-dev/phoenix-desktop/releases/download/prod-app-v3.6.4/phoenix-code_3.6.4_amd64_linux_bin-GLIBC-2.31.tar.gz with GLIBC version 2.31
Found a new best match: https://github.com/phcode-dev/phoenix-desktop/releases/download/prod-app-v3.6.4/phoenix-code_3.6.4_amd64_linux_bin-GLIBC-2.35.tar.gz with GLIBC version 2.35
Downloading the compatible binary from https://github.com/phcode-dev/phoenix-desktop/releases/download/prod-app-v3.6.4/phoenix-code_3.6.4_amd64_linux_bin-GLIBC-2.35.tar.gz...
/tmp/tmp.y2bvxw0tSX/phoenix-code.tar.gz       100%[=================================================================================================>]  73,70M  12,7MB/s    in 5,9s    
Downloading the icon...
/tmp/tmp.y2bvxw0tSX/icon.png                  100%[=================================================================================================>]  41,14K  --.-KB/s    in 0,02s   
Extracting the binary to /tmp/tmp.y2bvxw0tSX...
./phoenix-code: error while loading shared libraries: libwebkit2gtk-4.0.so.37: cannot open shared object file: No such file or directory
/tmp/tmp.y2bvxw0tSX/latest.json               100%[=================================================================================================>]   1,86K  --.-KB/s    in 0s      
Ubuntu 24.04 LTS is not currently supported by this installation script.
Please use an earlier version of Ubuntu for the time being. Check back later for updates.
```

Thank you,
Gaurav

**Comment by charlypa** [comment]
# Issue Resolution for Ubuntu 24.04

We are working to fix this as Ubuntu 24.04 removed support for `libwebkit2gtk-4.0.so.37`.

## Workaround

As a workaround, run these commands before running the install script again:

```bash
cd /usr/lib/x86_64-linux-gnu/
sudo ln -s /snap/gnome-42-2204/176/usr/lib/x86_64-linux-gnu/libwebkit2gtk-4.0.so libwebkit2gtk-4.0.so.37
sudo ln -s /snap/gnome-42-2204/176/usr/lib/x86_64-linux-gnu/libjavascriptcoregtk-4.0.so libjavascriptcoregtk-4.0.so.18
sudo ln -s /snap/gnome-42-2204/176/usr/lib/x86_64-linux-gnu/libwpe-1.0.so.1 libwpe-1.0.so.1
sudo ln -s /snap/gnome-42-2204/176/usr/lib/x86_64-linux-gnu/libWPEBackend-fdo-1.0.so.1 libWPEBackend-fdo-1.0.so.1
sudo ln -s /snap/gnome-42-2204/176/usr/lib/x86_64-linux-gnu/libicui18n.so.70 libicui18n.so.70
sudo ln -s /snap/gnome-42-2204/176/usr/lib/x86_64-linux-gnu/libicuuc.so.70 libicuuc.so.70
sudo ln -s /snap/gnome-42-2204/176/usr/lib/x86_64-linux-gnu/libicudata.so.70 libicudata.so.70
sudo ln -s /snap/gnome-42-2204/176/usr/lib/x86_64-linux-gnu/webkit2gtk-4.0 webkit2gtk-4.0
```

## Additional Information

Could you please let us know if you have done a fresh installation of Ubuntu 24.04 or did an upgrade from Ubuntu 23.04 as well?

**Comment by charlypa** [comment]
https://github.com/tauri-apps/tauri/issues/9662

**Comment by ghost** [comment]
@charlypa thank you and the tauri app work around will not work and i have done with both the developmental version and also the static compiled version. I did a upgrade from 23.04 and not a fresh install to see and going to do a complete fresh install in a coming week or so. Right now using the Kubuntu and the fresh install will be the ubuntu only and not with any type of flavour. 

Gaurav

**Reply by charlypa** [reply]
@gauravcodepro 
we are working to fix it

**Reply by charlypa** [reply]
If Kubuntu is based on ubuntu 23.04 Phoneix code  wont work there as well as we cant find the required dependencies

**Comment by ghost** [comment]
why not, it will work and it is working. see this i am writing the code right now also and it is showing well. I am using the Kubuntu 24.04 LTS for now and since the Kubuntu is only with the KDE installed, so it means it already supports the native libgtk widgets. 

![Screenshot from 2024-05-26 08-18-32](https://github.com/phcode-dev/phoenix/assets/19677953/d45e02df-525e-4fe1-810a-7730b3725a12)

Thank you,
Gaurav

**Reply by charlypa** [reply]
Ubutntu 24.04 has `libwebkit2gtk-4.1-0` however for Phoneix code we need `libwebkit2gtk-4.0-37`

**Reply by abose** [reply]
If you upgraded from 23, PhCode should work on it.

On fresh install, we are working with Tauri to make it work in Ubuntu 24. In the mean time, we are checking if we can do some fix at our end. https://github.com/tauri-apps/tauri/issues/9662

**Comment by ghost** [comment]
if other users are having problem then easy way is not to use the virtualbox till the tauri app gets ready. The best way is to install the distorbox and then use the minimal version of the ubuntu and then get the gtk widget and install phoneix. It works seamlessly. 

https://github.com/89luca89/distrobox?tab=readme-ov-file#quick-start
```
distrobox create -i ubuntu:20.04
```
Thanks
Gaurav

**Comment by ghost** [comment]
thank you and any guess by what time, the tauri based version shall be available for the Ubuntu 24.04 LTS. Thank you.

**Reply by abose** [reply]
As soon as tauri makes it available, please comment on the tauri issue too so that they would prioritize it. From our side we are investigating if we can do something from our end, mostly we are targeting june/max july.

**Comment by ghost** [comment]
Thank you and i just wanted to know as i am going to install a new system for myself and i use this one for the quick checks otherwise i use jetbrains and vscode all the time. Thank you

**Reply by charlypa** [reply]
@gauravcodepro 
We have updated our install script to support ubuntu 24.04. Please check it and see if it works for you

**Comment by ghost** [comment]
Thank you and this will enable me to complete my new system up timely with in this week.
